pkgs/os-specific/linux/nxp-pn5xx/default.nix

```
@@ -0,0 +1,41 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  fetchpatch,
+  kernel,
+  kmod,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "nxp-pn5xx";
+  version = "0.4-unstable-2024-12-27-${kernel.version}";
+
+  src = fetchFromGitHub {
+    owner = "jr64";
+    repo = "nxp-pn5xx";
+    rev = "127497649ba8489900ef30611412762846ec47d7";
+    hash = "sha256-z4pBkKKNvk8A4bDIhW/HTAIm50hKkpBLT1E87cF0KIs=";
+  };
+
+  nativeBuildInputs = [ kmod ] ++ kernel.moduleBuildDependencies;
+
+  makeFlags = kernel.makeFlags ++ [
+    "KERNELRELEASE=${kernel.modDirVersion}"
+    "BUILD_KERNEL_PATH=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build"
+    "INSTALL_MOD_PATH=$(out)/lib/modules/${kernel.modDirVersion}"
+  ];
+
+  postInstall = ''
+    mkdir -p $out/etc/udev/rules.d
+    echo 'SUBSYSTEM=="misc", KERNEL=="pn544", MODE="0666", GROUP="dialout"' > $out/etc/udev/rules.d/99-nxp-pn5xx.rules
+  '';
+
+  meta = with lib; {
+    description = "NXP's NFC Open Source Kernel mode driver with ACPI configuration support";
+    homepage = "https://github.com/jr64/nxp-pn5xx";
+    license = licenses.gpl2Only;
+    maintainers = with maintainers; [ stargate01 ];
+    platforms = platforms.linux;
+  };
```

Doesn't build on 6.13.
```suggestion
    broken = kernel.kernelAtLeast "6.13";
  };
```

---

Thank you for bringing this up, this is a upstream problem with the kernel module which I will address before merging this PR.

---

Thanks to the upstream module patch, the module now properly builds and runs on kernel 6.13 as well. The issue was the (legitimate) coexistence of `CONFIG_OF` and `CONFIG_ACPI` in the kernel config, the module now no longer uses compile time configuration but instead properly does runtime devicetree / ACPI detection.